### DISCUSSION TITLE
Use slog from golang's new stdlib instead of zap

### DISCUSSION THREAD

**Discussion by kishaningithub** [jeroenrinzema/psql-wire#60] [Ideas]
slog (Structured LOGging) is now getting built right into golang (release targeted for 1.21). The API as such is inspired by zap to the transition should be straight forward. 

Pros
- The consumers of this library do not have to get tied up with zap logger and they are free to plug and play an slog implementation of choice (stdlib, zap etc) stdlib should mostly suffice the needs of all consumers IMO.

The implementation is currently  available at https://pkg.go.dev/golang.org/x/exp/slog

More info
- https://youtu.be/gd_Vyb5vEw0
- https://github.com/golang/go/issues/56345

**Comment by jeroenrinzema** [comment]
Hi Thanks for the heads up! I like the idea of using `slog` inside the `psql-wire` project. I do not want to make this refactor at the moment since other projects that I am currently working on are relying on the `zap` logger inside the `psql-wire` project. I would like to make this change once `slog` has moved out of the experimental package and I have migrated the other applications.

**Reply by kishaningithub** [reply]
@jeroenrinzema Go 1.21 is now out and it has the [slog package](https://pkg.go.dev/log/slog) :-) 

https://go.dev/blog/go1.21

BTW it also has the [slog test package](https://go.dev/doc/go1.21#slogtest) to facilitate testing of logs

**Reply by jeroenrinzema** [reply]
@kishaningithub thanks for the heads up! We are currently planning to migrate our main project to use slog. Feel free to open a PR to migrate the `psql-wire` repo. I will update the go version in `psql-wire` to 1.21.0.

**Comment by jeroenrinzema** [comment]
Hi @kishaningithub just wanted to let you know that I have just replaced zap with slog in the `v0.7.0` release.
